src/lib.rs

```diff
@@ -5,52 +5,42 @@ use serde::de::DeserializeOwned;
 use serde::Serialize;
 use std::marker::Unpin;
 
-// Transport configurtatiion trait
+/// Transport configuration trait.
 pub trait TransportConfiguration<Data> {
-    // creates new transport configuration with specified network
-    // address for incoming messages listener
-    fn new(set_bind_net_addr: String) -> Self;
+    /// Creates a new transport configuration with the specified network address for incoming
+    /// message listener.
+    fn new(set_bind_net_addr: &str) -> Result<Box<Self>>;
```

Comment:
What is the reson to change result type to `Result<Box<Self>>` ?

Reply:
Because `Self` is a trait object which is not `Sized` and therefore cannot be passed by value inside the `Result`, only by a fat pointer.

You do need to return a `Result` and any possible errors. It's not an option to `panic` inside those methods.

I generally recommend to avoid trait objects because they lead to much slower running code compared to static dispatch.

Reply:
You can add `Sized` bound to `Self, no need to put it into `Box`.

Reply:
No. Sizes of objects of instances of `Transport` are not known at compile time.

Reply:
@Maxime2 I see you added the annotation [yourself](https://github.com/Fantom-foundation/libtransport/blob/c93dd87fb57232738018932c699dbb8535df5baf/src/lib.rs#L19). It's not going to help because the compiler cannot determine the size of `TcpTransportCfg`. So could you please remove it.

Reply:
Agreeing with @vkomenda here. We've got to be extra careful when writing Rust code to handle paths where sizes are unknown until runtime.

Reply:
I decided to require `Sized` for `TransportConfiguration` and have no such requirement in `Transport`. That hasn't presented any problem yet. If it does, we can go back to boxing the returned value of `TransportConfiguration::new` and this is not going to introduce overhead because configuration steps are infrequent.